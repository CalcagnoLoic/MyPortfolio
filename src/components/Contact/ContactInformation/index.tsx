import Paragraph from "../../../typographies/Paragraph";

type ContactInformationProps = {
  icon: string;
  content: string | JSX.Element;
};

const Component = ({ icon, content }: ContactInformationProps) => (
  <div className="mb-3 flex justify-center gap-5 rounded-3xl bg-gallery p-5 text-center text-tuatara md:text-start">
    <Paragraph kind="span" content={icon} css="self-center text-xl" />
    <Paragraph kind="p" content={content} css="" />
  </div>
);

export default Component;
