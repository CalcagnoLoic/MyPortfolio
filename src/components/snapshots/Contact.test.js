import renderer from "react-test-renderer";
import Contact from "../Contact";

it("render correctly Contact component", () => {
    const componentContact = renderer.create(<Contact />).toJSON();
    expect(componentContact).toMatchSnapshot();
});
