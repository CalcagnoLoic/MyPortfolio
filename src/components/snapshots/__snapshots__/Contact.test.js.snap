// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly Contact component 1`] = `
<section
  className="lg:py-10"
>
  <h2
    className="text-center text-main-color pt-10 text-3xl lg:text-4xl font-bold font-[sofia] underline"
    id="contact"
  >
    Get in touch
  </h2>
  <div
    className="flex flex-col md:flex-row justify-center items-center mx-2 sm:mx-5 lg:mx-10"
  >
    <div
      className="self-center mx-10 md:pl-5 lg:mx-5 lg:pl-20"
    >
      <h2
        className="text-center md:text-start pt-5 md:pt-0 text-main-color"
      >
        If you want more information about me or my work, don't hesitate to contact me with this contact form or directly on my email address.
         
      </h2>
      <h2
        className="text-center md:text-start mb-10 text-main-color"
      >
        I will reply as soon as possible.
      </h2>
      <div
        className="flex flex-col gap-5"
      >
        <div
          className="flex gap-5 text-center justify-center md:text-start border-2 p-5 rounded-3xl bg-gray-100 border-white"
        >
          <svg
            className="self-center"
            fill="currentColor"
            height="1.25rem"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            version="1.1"
            viewBox="0 0 17 17"
            width="1.25rem"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g />
            <path
              d="M0 2v13h17v-13h-17zM8.494 9.817l-6.896-6.817h13.82l-6.924 6.817zM5.755 8.516l-4.755 4.682v-9.383l4.755 4.701zM6.466 9.219l2.026 2.003 1.996-1.966 4.8 4.744h-13.677l4.855-4.781zM11.201 8.555l4.799-4.725v9.467l-4.799-4.742z"
            />
          </svg>
          <p
            className="self-center text-main-color"
          >
            calcagnoloic93@gmail.com
          </p>
        </div>
        <div
          className="flex gap-5 text-center justify-center md:text-start border-2 p-5 rounded-3xl bg-gray-100 border-white"
        >
          <svg
            className="self-center"
            fill="currentColor"
            height="1.25rem"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1.25rem"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17.5,6.5 L23,9 L23,22 L16,19 L8,22 L1,19 L1,6 L6,8 M16,19 L16,12 M8,22 L8,12 M12,16.2727273 C12,16.2727273 6,11.5 6,7 C6,3.25 9,1 12,1 C15,1 18,3.25 18,7 C18,11.5 12,16.2727273 12,16.2727273 Z M13,7 C13,6.44766667 12.5523333,6 12,6 C11.4476667,6 11,6.44766667 11,7 C11,7.55233333 11.4476667,8 12,8 C12.5523333,8 13,7.55233333 13,7 Z"
              fill="none"
              stroke="#000"
              strokeWidth="2"
            />
          </svg>
          <p
            className="self-center text-main-color"
          >
            Belgium, Hainaut, 7170 La Hestre
          </p>
        </div>
      </div>
    </div>
    <div>
      <form
        action=""
        className="flex flex-col my-5 py-8 sm:px-5 lg:pr-16 lg:px-16"
        onSubmit={[Function]}
      >
        <div
          className="flex flex-col lg:flex-row lg:gap-5"
        >
          <div
            className="flex flex-col sm:gap-1 sm:mb-8 mb-2"
          >
            <label
              className="mb-1 font-semibold text-main-color"
            >
              Name
               
              <span
                className="text-red-element"
              >
                *
              </span>
               
              :
               
            </label>
            <input
              className="rounded-md p-2 w-72 md:w-80 border-2"
              name="name"
              placeholder="Name"
              required={true}
              type="text"
            />
          </div>
          <div
            className="flex flex-col sm:gap-1 sm:mb-8 mb-2"
          >
            <label
              className="mb-1 font-semibold text-main-color"
            >
              Email
               
              <span
                className="text-red-element"
              >
                *
              </span>
               
              :
               
            </label>
            <input
              className="rounded-md w-72 p-2 md:w-80 border-2"
              name="reply_to"
              placeholder="Email"
              required={true}
              type="email"
            />
          </div>
        </div>
        <div
          className="flex flex-col sm:gap-1 sm:mb-8 mb-2"
        >
          <label
            className="mb-1 font-semibold text-main-color"
          >
            Message
             
            <span
              className="text-red-element"
            >
              *
            </span>
             :
             
          </label>
          <textarea
            className="rounded-md w-72 p-2 md:w-80 lg:w-full border-2"
            cols="30"
            name="message"
            placeholder="Message"
            required={true}
            rows="4"
          />
          <span
            className="text-red-element"
          >
            * Required Fields
          </span>
        </div>
        <div
          className="grid place-content-center md:block"
        >
          <input
            className="text-white font-bold p-4 hover:bg-red-element rounded-xl cursor-pointer hover:text-white bg-main-color duration-1000 text-lg mt-3 w-full md:w-1/2"
            type="submit"
            value="Submit"
          />
        </div>
      </form>
    </div>
  </div>
</section>
`;
